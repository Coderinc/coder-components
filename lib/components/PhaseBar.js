import React from 'react';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${props =>
    props.active
      ? props.theme.activeBackground
      : props.theme.primaryBackground};
  color: ${props => props.theme.primaryFont};

  display: flex;
  align-items: center;

  box-sizing: border-box;
  width: 100%;
  padding: 0.5rem;
  border-radius: 0.25rem;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const Fill = styled.div`
  flex: 1;
`;

export const PhaseBar = ({ label, phase, ...props }) => (
  <BarContainer {...props}>
    <Paragraph>{label}</Paragraph>
    <Fill />
    <Chip label={phase} />
  </BarContainer>
);

export default PhaseBar;
