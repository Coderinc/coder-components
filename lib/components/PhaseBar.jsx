import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Paragraph from '../elements/Paragraph';
import Chip from '../elements/Chip';

const BarContainer = styled.div`
  background-color: ${({ active, theme }) => (active ? theme.activeBackground : theme.offsetBackground)};
  color: ${({ theme }) => theme.primaryFont};

  display: flex;
  align-items: center;

  box-sizing: border-box;
  width: 100%;
  margin: 0.25rem 0;
  padding: 0.5rem;
  border-radius: 0.25rem;

  &::before {
    font-size: 1.5rem;
    font-family: coder-icons;
    font-weight: normal;
    content: '\f108';
    margin-right: 0.5rem;
  }
`;

const Fill = styled.div`
  flex: 1;
`;

const PhaseBar = ({
  label, phase, value, onClick, ...props
}) => (
  <BarContainer {...props} onClick={onClick ? () => onClick(value) : undefined}>
    <Paragraph>
      {label}
    </Paragraph>
    <Fill />
    <Chip label={phase} />
  </BarContainer>
);

PhaseBar.propTypes = {
  label: PropTypes.string.isRequired,
  phase: PropTypes.string.isRequired,
  // eslint-disable-next-line
  value: PropTypes.any,
  onClick: PropTypes.func,
};

PhaseBar.defaultProps = {
  value: 0,
  onClick: undefined,
};

export default PhaseBar;
