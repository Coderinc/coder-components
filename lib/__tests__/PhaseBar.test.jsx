import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import 'jest-styled-components';
import PhaseBar from '../components/PhaseBar';

describe('PhaseBar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<PhaseBar label="Lorem Ipsum" phase="Design" />)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('fires on click', () => {
    const onClick = sinon.spy();
    const component = mount(<PhaseBar label="Lorem" phase="T" value={1} onClick={onClick} />);
    component.find('BarContainer').simulate('click');
    expect(onClick.calledOnce).toEqual(true);
  });
})