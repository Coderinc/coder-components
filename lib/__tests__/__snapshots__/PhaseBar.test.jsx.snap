// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseBar renders correctly 1`] = `
.c1 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: lighter;
}

.c3 {
  padding: 0.25rem 0.5rem;
  margin: 0.25rem;
  border-radius: 0.125rem;
  font-size: 0.625rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  margin: 0.25rem 0;
  padding: 0.5rem;
  border-radius: 0.25rem;
}

.c0::before {
  font-size: 1.5rem;
  font-family: coder-icons;
  font-weight: normal;
  content: '\\f108';
  margin-right: 0.5rem;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    Lorem Ipsum
  </p>
  <div
    className="c2"
  />
  <div
    className="c3"
    label="Design"
  >
    Design
  </div>
</div>
`;
